### PR TITLE
chore(ssa refactor): Handle Brillig calls

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

## Problem\*

Resolves 

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: crates/noirc_evaluator/src/brillig/brillig_ir.rs [hunk 1]
```diff
@@ -543,6 +536,108 @@ impl BrilligContext {
...
+    /// Utility method to perform a binary instruction with a constant value
+    pub(crate) fn usize_op(
+        &mut self,
+        destination: RegisterIndex,
+        op: BinaryIntOp,
+        constant: usize,
+    ) {
+        let const_register = self.make_constant(Value::from(constant));
+        self.binary_instruction(
+            destination,
```
Comment (kevaundray) [inline crates/noirc_evaluator/src/brillig/brillig_ir.rs:590]: This is a bug -- we are adding destination to iteself into `const_register`

**Comment by ludamad** [comment]
Ah, my bad with `usize_op` - I had started with brillig to list destination first as that is intel x86 convention so my brain went there

**Comment by kevaundray** [comment]
Changes look okay to me -- going to give it a once over again tomorrow, then we can open an issue re jumps in function calls

**Comment by kevaundray** [comment]
@sirasistant Looks good to me, can you approve?

**Review by sirasistant — DISMISSED** [review]
sure!

**Review by sirasistant — APPROVED** [review]

**Comment by sirasistant** [comment]
Had to add it to the merge queue again because seems like some CI step stalled and didn't give output

**Comment by ludamad** [comment]
I keep looking and being surprised this isnt in master :D

**Pull request merged**